create-hpc-cluster-linux-cn/azuredeploy.json

```diff
@@ -1,468 +1,519 @@
+      }
+    },
+    "computeNodeNumber": {
+      "type": "int",
+      "defaultValue": 2,
+      "metadata": {
+        "description": "The number of the compute nodes"
+      }
+    },
+    "computeNodeVMSize": {
+      "type": "string",
+      "defaultValue": "Standard_A3",
+      "allowedValues": [
+        "Standard_A1",
+        "Standard_A2",
+        "Standard_A3",
+        "Standard_A4",
+        "Standard_A5",
+        "Standard_A6",
+        "Standard_A7",
+        "Standard_A8",
+        "Standard_A9",
+        "Standard_A10",
+        "Standard_A11",
+        "Standard_D1",
+        "Standard_D2",
+        "Standard_D3",
+        "Standard_D4",
+        "Standard_D11",
+        "Standard_D12",
+        "Standard_D13",
+        "Standard_D14"
+      ],
+      "metadata": {
+        "description": "The VM role size of the compute nodes"
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "The user name of the administrator"
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "The password of the administrator"
+      }
+    },
+    "headNodePostConfigScript": {
+      "type": "string",
+      "defaultValue": " ",
+      "metadata": {
+        "description": "Optional, specify the script url and command line if you want to run your custom script on the head node after it is configured. The script url must be public available, and you can also specify arguments for the script, for example 'http://www.consto.com/mypostscript.ps1 -Arg1 arg1 -Arg2 arg2'."
+      }
+    }
+  },
+  "variables": {
+    "apiVersion": "2015-05-01-preview",
+    "storageAccountName": "[toLower(concat(parameters('clusterName'), 'sa'))]",
+    "cnStorageAccountPrefix": "[concat(variables('storageAccountName'), 'cn')]",
+    "cnStorageAccountNumber": "[add(div(parameters('computeNodeNumber'), 10), 1)]",
+    "storageAccountId": "[resourceId('Microsoft.Storage/storageAccounts',variables('storageAccountName'))]",
+    "virtualNetworkName": "[concat(parameters('clusterName'), 'VNet')]",
+    "addressPrefix": "10.0.0.0/16",
+    "subnet1Name": "Subnet-1",
+    "subnet1Prefix": "10.0.0.0/24",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
+    "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnet1Name'))]",
+    "publicIPAddressType": "Dynamic",
+    "publicIPAddressName": "[concat(parameters('clusterName'), '-PublicIP-VM')]",
+    "publicIPAddressNameHN": "[concat(variables('publicIPAddressName'),'-HN')]",
+    "availabilitySetName": "[concat(parameters('clusterName'), 'AVSet')]",
+    "dnsName": "[concat(parameters('clusterName'), 'ep')]",
+    "nicName": "[concat(parameters('clusterName'),'-nic')]",
+    "nicNameHN": "[concat(variables('nicName'),'-hn')]",
+    "vmSizeRDMAStates": {
```

Comment:
out of curiosity, how does this object work? Some states are enabled and some are disabled